src/AwsMetadata.js

```diff
@@ -53,7 +54,7 @@ export default class AwsMetadata {
       this.lookupMetadataKey(`network/interfaces/macs/${results.mac}/vpc-id`, (err, vpcId) => {
         results['vpc-id'] = vpcId;
         this.logger.debug('Found Instance AWS Metadata', results);
-        resultsCallback(results);
+        resultsCallback(_(results).omitBy(_.isUndefined).omitBy(_.isNull).value());
```

Comment:
I'd prefer we do this with native JS array/object features instead of pulling in a lodash dependency.